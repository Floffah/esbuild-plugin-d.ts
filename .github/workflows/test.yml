name: Test

on: [push, pull_request]

jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
      - uses: oven-sh/setup-bun@v1
      - run: bun install --frozen-lockfile
      - run: bun run test
