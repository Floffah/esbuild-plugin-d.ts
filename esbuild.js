const {build} = require("esbuild");

const dev = process.argv.includes("--dev") || process.argv.includes("-d");

build({
    bundle: true,
    target: "node16.0",
    platform: "node",
    watch: dev,
    color: true,
    logLevel: "info",
    minify: !dev,
    minifyIdentifiers: !dev,
    minifySyntax: !dev,
    minifyWhitespace: !dev,
    format: "cjs",
    entryPoints: ["./src/index.ts"],
    outfile: "./dist/index.cjs",
    external: ["esbuild", "typescript", "chalk"]
})
