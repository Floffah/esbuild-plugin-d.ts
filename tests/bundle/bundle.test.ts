import { dtsPlugin } from "../../dist";
import { distDir, readOutputFile } from "../_utils";
import { expect, test } from "bun:test";
import { build } from "esbuild";
import { resolve } from "path";

test("Basic config", async () => {
    const tsconfig = resolve(__dirname, "./tsconfig.json");

    await build({
        plugins: [dtsPlugin({ tsconfig, experimentalBundling: true })],
        entryPoints: [
            resolve(__dirname, "./inputs/bundle.ts"),
            resolve(__dirname, "./inputs/secondBundle.ts"),
        ],
        outdir: distDir,
        tsconfig,
        bundle: true,
    });

    expect(readOutputFile("bundle")).toMatchSnapshot();
    expect(readOutputFile("secondBundle")).toMatchSnapshot();
});
