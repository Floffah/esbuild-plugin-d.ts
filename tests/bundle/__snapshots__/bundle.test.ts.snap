// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic config 1`] = `
"// Generated by dts-bundle-generator v9.5.1

export declare function a(): string;
export declare function b(): string;

export {};
"
`;

exports[`Basic config 2`] = `
"// Generated by dts-bundle-generator v9.5.1

export declare function b(): string;
export declare function c(): string;

export {};
"
`;
